packs/classfeatures/deadly-strike.json

```diff
@@ -24,7 +24,41 @@
             "remaster": false,
             "title": "Pathfinder Guns & Gears"
         },
-        "rules": [],
+        "rules": [
+            {
+                "definition": [
+                    "item:id:{actor|flags.pf2e.trackedItems.weaponInnovation}"
+                ],
+                "key": "AdjustStrike",
+                "mode": "add",
+                "property": "weapon-traits",
+                "value": "deadly-d8"
+            },
+            {
+                "definition": [
+                    "item:id:{actor|flags.pf2e.trackedItems.weaponInnovation}"
+                ],
+                "key": "AdjustStrike",
+                "mode": "add",
+                "predicate": [
+                    "item:trait:deadly-d6"
+                ],
+                "property": "weapon-traits",
+                "value": "deadly-d10"
+            },
```

Comment:
Does this add both traits? If so we need another set of REs to remove the old trait (change the priority to 1 higher than the add RE has)